src/typeahead/typeahead.ts

```diff
@@ -302,6 +302,7 @@ export class NgbTypeahead implements ControlValueAccessor,
 
     if (!defaultPrevented) {
       this.writeValue(result);
+      this._userInput = result;
```

Comment:
This looks odd... `_userInput` should be a string while here we would be assigning an item which is potentially a complex object. Why do we need this assignment?

If you need to store `_userInput` you should probably store a stringified value inside the `this.writeValue` method.